debian/snapd.postrm

```diff
@@ -3,7 +3,19 @@
 set -e
 
 if [ "$1" = "purge" ]; then
-   # FIXME: should we try to remove all snaps, mount points, services,
-   #        mount units that got created as well?
-   rm -f /var/lib/snapd/state.json
-fi
\ No newline at end of file
+
+    for snap in /var/lib/snapd/snaps/*.snap; do
+        base="$(basename $snap)"
+        # mount units
+        mangledbase="$(echo $base|tr _ -)"
+        mountunit="$(systemd-escape -p "${mangedbase%%.snap}".mount)"
+        rm -f /etc/systemd/system/snap-$mountunit
+        rm -f /etc/systemd/system/multi-user.target.wants/snap-$mountunit
+
+        # FIXME: umount snaps
```

Comment:
Yeah, I think we want to do that before we remove these unit files. Seems strange to remove and leave the content behind.

Note we'll need to:

1. Stop service units
1. Stop mount units
1. Remove service and mount units
1. Remove all content

The tests/lib/reset.sh actually holds exactly this logic, except your method of finding out mount units is nicer above.